### PR TITLE
Fixed bad requests (url="http:/")

### PR DESCRIPTION
**Pull request by mvrozanti**
Hi there
Great tool you have made
I made some changes that better suited my use case (I gotta pipe the output so it has to be really clean on the least verbose level)
I also fixed (?) a strange error of a malformed request that was being created through a `Resource` object with a bad url.

- replaced `logger.warn` for `logger.warning`
- removed every unnecessary `sudo pip` instruction
- optional dependencies logging is now dependant on verbose level
- error logging in download function now depends on verbose level

?*: there's probably a better way to do this; the url regex is too simplified and may give some false positives.

### CODE REVIEW

**Comment by dhondta** [comment]
Hi !
Thank you.

A few remarks on your modifications : 
1. `logger.warning`: You're right, `logger.warn` has been deprecated.
2. `sudo pip`: A bit cosmetic, but it's safer not to use `sudo`, so you're right.
3. `re.match(...)` at line 345: With this, you will break an important part of the features as the `url` parameter is also aimed to support raw data (see lines 562 and 577).
4. Import messages depending on verbose level: The script must fail if BeautifulSoup is not installed but we could eventually put the message about `coloredlogs` (at line 59) where you put the blocks. But I can't accept your changes in this state.
5. Exception catching of failing resources: This is not the aim to bypass any resource-related exception. If you discovered weird exceptions while using the tool, I would ask you to open issues instead of silenting any of these.

In conclusion, I can't accept your PR, i.e. points 3 to 5.
If you want, you can rewrite your changes with a commit for the points 1 and 2. You can also eventually adapt the point 4 in another commit.
